canonical-web-and-design/maas-ui
# refactor(ui): update slice generation to only generate reducers

## Done

- Generate the common reducers instead of generating the whole slice as this provides much better type
support.

## How to review this PR

- The main changes are in [utils/slice.ts](https://github.com/canonical-web-and-design/maas-ui/pull/2694/files#diff-79f2f5b73d03afbd8158f5b7780ab97de74cb23247511bc89f56ab8fb3531445). Instead of generating the entire slice in the util, this just generates the common reducers.
- The next changes are updating the slice files for each model (e.g. [vlan/slice.ts](https://github.com/canonical-web-and-design/maas-ui/pull/2694/files#diff-d04eab3dac53d9dab4c839b45be5c5d664633f07d524bfe6368d9626ccc5ee7d). The slice is now defined in the file and common reducer util is used to added the reducers. This results in less code and better type support.
- The rest of the updates are largely fixing issues that weren't visible before.
- The status reducers generated by `generateStatusHandlers` still don't have good type support and will need to be revisited at some point.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Use redux dev tools to check that the data is being loaded into state.
- Try viewing and performing actions on machines and pods as they have the most complicated slices.

## Fixes

Fixes: #2049.
